### ISSUE TITLE
[12.0] stock: Error deleting stock.picking.type

### ISSUE THREAD

**Issue by elperrofarelo** [OCA/OpenUpgrade#1925]
Impacted versions:
12
Steps to reproduce:
Migration done successfully.
After restart and logging in while upgrading stock app (version 12) comes following error
Current behavior:
2019-07-11 06:39:42,163 6161 DEBUG odoo11 odoo.sql_db: query: SELECT count(1) FROM "ir_model_data" WHERE (((("ir_model_data"."model" = 'stock.picking.type')  AND  ("ir_model_data"."res_id" = 2))  AND  ("ir_model_data"."id" != '8444'))  AND  (TRUE OR "ir_model_data"."id" IS NULL)) 
2019-07-11 06:39:42,165 6161 INFO odoo11 odoo.addons.base.models.ir_model: Deleting 2@stock.picking.type (stock.picking_type_out) 
2019-07-11 06:39:42,166 6161 DEBUG odoo11 odoo.sql_db: query: SELECT id FROM "stock_picking_type" WHERE id IN (2) 
2019-07-11 06:39:42,170 6161 DEBUG odoo11 odoo.sql_db: query: SELECT "ir_property".id FROM "ir_property" WHERE ("ir_property"."res_id" IS NULL   AND  ("ir_property"."value_reference" in ('stock.picking.type,2'))) ORDER BY "ir_property"."id"   
2019-07-11 06:39:42,173 6161 DEBUG odoo11 odoo.sql_db: query: SELECT "ir_property".id FROM "ir_property" WHERE ("ir_property"."res_id" in ('stock.picking.type,2')) ORDER BY "ir_property"."id"   
2019-07-11 06:39:42,175 6161 DEBUG odoo11 odoo.sql_db: query: DELETE FROM stock_picking_type WHERE id IN (2) 
2019-07-11 06:39:42,204 6161 ERROR odoo11 odoo.sql_db: bad query: b'DELETE FROM stock_picking_type WHERE id IN (2)'
ERROR: null value in column "picking_type_id" violates not-null constraint
DETAIL:  Failing row contains (1, WH: Stock -> Customers, t, propagate, null, pull, 20, 1, 9, 12, 3, make_to_stock, 10, null, 0, null, t, 1, null, 1, 2017-10-24 08:00:58.598387, 1, 2017-10-24 10:04:39.768136, manual).
CONTEXT:  SQL statement "UPDATE ONLY "public"."stock_rule" SET "picking_type_id" = NULL WHERE $1 OPERATOR(pg_catalog.=) "picking_type_id""
 
Expected behavior:

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
In your DB you have some scrambled data that makes the regular Odoo update process to remove that data, but that seems to not have something to do with OpenUpgrade. Check your data o remove the linked XML-ID for avoiding the removal.

**Comment by elperrofarelo** [comment]
I've compared with a new DB V12 and the data are similar. Can you please tell me where are the code lines in OpenUpgrade that are trying to delete:
Deleting 2@stock.picking.type (stock.picking_type_out)
I've checked the code and I can't find them

**Comment by pedrobaeza** [comment]
As said, this is done by regular Odoo update process, not OpenUpgrade.

**Comment by elperrofarelo** [comment]
Must I then delete the stock.picking.type and stock.rule tables and upgrade again??

**Comment by pedrobaeza** [comment]
I would bet first for doing a complete update on the previous version, and then see the problems.

**Comment by elperrofarelo** [comment]
Thanks. I'll try

**Comment by pedrobaeza** [comment]
I have faced this today migrating an instance, so reopening while I investigate if this is something general.

**Comment by pedrobaeza** [comment]
OK, indeed there's a problem in OpenUpgrade. It's being fixed in #1927, but you can run this SQL in your migrated DB:
```sql
UPDATE ir_model_data
SET noupdate = True
WHERE module = 'stock'
AND name IN (
    'stock_location_stock',
    'stock_location_company',
    'stock_location_output',
    'location_pack_zone',
    'picking_type_internal',
    'picking_type_in',
    'picking_type_out'
);
```
and then proceed with the regular Odoo update process for not needing to migrate again.

**Comment by elperrofarelo** [comment]
Thanks in any case, I've repaired everything manually.
